.github/workflows/build-docs.yml

```diff
@@ -0,0 +1,44 @@
+# This workflow installs the package and builds the docs
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: build the docs
+
+on:
+  push:
+    branches: [ 'main' ]
+  pull_request:
+    branches: [ '**' ]
+
+jobs:
+  pytest:
+    strategy:
+      matrix:
```

Comment:
- Since this only needs one job on one OS/version, could simply use [`runs-on:`](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on).
- `fail-fast:` is only relevant when there are multiple jobs.

Reply:
I slightly prefer to keep this implementation so that it is easy to add other OS's/versions later.